bin/cle-test

```diff
@@ -0,0 +1,349 @@
+    exit(255);
+}
+# functions
+sub get_sample_from_subject_mapping {
+    my ($anp, $tag) = @_;
+    my $mapping = Genome::Config::AnalysisProject::SubjectMapping->get(analysis_project => $anp,
+        tags => [$tag]);
+    my @bridges = grep {$_->label eq 'tumor_sample'} $mapping->subject_bridges;
+    my $bridge = $bridges[0];
+    return $bridge->subject;
+}
+
+sub get_timeout {
+    my $timeout_hours = 14;
+    my $timeout_seconds = $timeout_hours * 3600;
+
+    return $timeout_seconds;
+}
+
+sub get_process_timeout {
+    my $timeout_hours = 14;
+    my $timeout_seconds = $timeout_hours * 3600;
+
+    return $timeout_seconds;
+}
+
+sub wait_for_build {
+    my $build = shift;
+    my $start_time = shift;
+    my $event = $build->the_master_event;
+    unless ($event) {
+        fail("Could not get the build's master event!\n");
+    }
+
+    my $timeout = get_timeout();
+    printf("Monitoring build (%s) until it completes or timeout "
+        . "of %s minutes is reached.\n\n", $build->id, $timeout / 60);
+
+    while (!grep { $event->event_status eq $_ } ('Succeeded',
+            'Failed', 'Crashed')) {
+        UR::Context->current->reload($event);
+        $build = UR::Context->current->reload('Genome::Model::Build', id => $build->id);
+
+        my $elapsed_time = time - $start_time;
+        if ($elapsed_time > $timeout) {
+            printf("Build (%s) timed out after %s minutes",
+                $build->id, $timeout / 60);
+            Library::send_timeout_mail();
+            build_view_and_exit($build);
+        }
+
+        sleep(30);
+    }
+}
+
+sub wait_for_process {
+    my $process = shift;
+
+    my $timeout = get_process_timeout();
+    printf("Monitoring process (%s) until it completes or timeout "
+        . "of %s minutes is reached.\n\n", $process->id, $timeout / 60);
+
+    my $start_time = time;
+    while (!grep { $process->status eq $_ } ('Succeeded', 'Crashed')) {
+        UR::Context->current->reload($process);
+
+        my $elapsed_time = time - $start_time;
+        if ($elapsed_time > $timeout) {
+            printf("Process (%s) timed out after %s minutes",
+                $process->id, $timeout / 60);
+            Library::send_timeout_mail();
+            process_view_and_exit($process);
+        }
+
+        sleep(30);
+    }
+}
+
+sub process_view_and_exit {
+    my $process = shift;
+    my $pv_command = Genome::Process::Command::View->create(
+        process => $process);
+    $pv_command->execute;
+    exit(255);
+}
+
+sub check_build_failure {
+    my $build = shift;
+
+    if ($build->status eq 'Succeeded') {
+        printf("Build status is %s.\n", $build->status);
+    } else {
+        Library::send_fail_mail();
+        build_view_and_exit($build);
+    }
+}
+
+sub diff_build {
+    my $build = shift;
+
+    printf('Starting diff (new build = %s)...', $build->id);
+
+    my @blessed_builds = Genome::Model::Build->get(id => [qw(
+    185d8bac3d7c4437b7ce9207dfadac7e
```

Comment:
I wonder if it might be better to have the blessed build IDs in the Jenkins configuration instead of in this repo.  (I know that's not how we do it with the regular model tests, either.)